packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -335,6 +335,8 @@ class MenuItem extends AutoControlledComponent<WithAsProp<MenuItemProps>, MenuIt
         e.stopPropagation()
         e.preventDefault()
       }
+    } else {
+      e.preventDefault()
```

Comment:
> * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.

So this action can be dispatched by `Enter` key, too. @sophieH29 is it correct to prevent default on it?

Reply:
We cannot do this, this broke the submenu behavior, try opening with keyboard some submenu and press enter on some action and you will see that the menus are not closed. Awkward that this was not caught by any test :(

Reply:
Actually, not sure whether it is broken, but we have to ensure that we will not break the behavior...

Reply:
@layershifter yes should be ok, no implications. We've done it for the most action handlers - on Enter/Space keys

Reply:
I don't know what the exact reason is but as @mnajdova wrote submenu now does not close on Enter/Space with `menuBehavior` or `menuAsToolbarBehavior`. In master it does not work for `menuBehavior` but it closes as expected when `menuAsToolbarBehavior` is used.

@lucivpav could you please investigate it before merging the PR?